openpype/hosts/maya/plugins/publish/extract_playblast.py

```diff
@@ -50,8 +51,37 @@ def process(self, instance):
                                ['override_viewport_options']
         )
         preset = lib.load_capture_preset(data=self.capture_preset)
-
+        # Grab capture presets from the project settings
+        capture_presets = self.capture_preset
+        # Set resolution variables from capture presets
+        width_preset = capture_presets["Resolution"]["width"]
+        height_preset = capture_presets["Resolution"]["height"]
+        # Set resolution variables from asset values
+        asset_width = instance.data.get("resolutionWidth")
+        asset_height = instance.data.get("resolutionHeight")
+        review_instance_width = instance.data.get("instanceWidth")
+        review_instance_height = instance.data.get("instanceHeight")
         preset['camera'] = camera
+
+        # Tests if project resolution is set,
+        # if it is a value other than zero, that value is
+        # used, if not then the asset resolution is
+        # used
+
+        if review_instance_width != 0:
+            preset['width'] = review_instance_width
+        elif width_preset == 0:
+            preset['width'] = asset_width
+        elif width_preset != 0:
+            preset['width'] = width_preset
+
+        if review_instance_height != 0:
+            preset['height'] = review_instance_height
+        elif height_preset == 0:
+            preset['height'] = asset_height
+        elif height_preset != 0:
+            preset['height'] = height_preset
```

Comment:
You should think about width and height as "pair" of values.
```suggestion
        if review_instance_width and review_instance_height:
            preset['width'] = review_instance_width
            preset['height'] = review_instance_height
        elif width_preset and height_preset:
            preset['width'] = asset_width
            preset['height'] = asset_height
        elif width_preset and height_preset:
            preset['width'] = width_preset
            preset['height'] = height_preset
```

Reply:
Thank you, this test is specifically sometimes testing for either values existing or not existing (for example, if it's a zero somewhere and not a zero elsewhere, so while I understand that I should be handling them both I'm not sure this should be how I approach it considering I'm trying to find if it's zero at some point or if it's not at another.